src/piece_picker.cpp

```diff
@@ -333,6 +333,12 @@ namespace libtorrent::aux {
 		*zero_prio = int(m_downloads[piece_pos::piece_zero_prio].size());
 	}
 
+    void piece_picker::set_sequential_start(piece_index_t const start_piece, piece_index_t const end_piece)
+    {
+		m_sequential_start = start_piece;
+		m_sequential_end = end_piece;
```

Comment:
it would probably make sense to scan `m_sequential_start` forward until you find the first piece we don't have, and likewise with `m_sequential_end`, scanning backwards.

That way you don't have to scan every time we try to pick a block. Similarly to how `m_cursor` and `m_reverse_cursor` is handled, you'd scan forward every time this piece finishes.